Zowe Desktop Documentation Viewer

ParthS007 opened the pull request.

ParthS007 committed:
Add WebClient using React framework to get the documentation details
Add nodeServer using Node and Express to send the documentation details to client

Signed-off-by: Parth Shandilya <parth1989shandilya@gmail.com>

ParthS007 committed:
Fixed requested changes
1. Added context logger
2. Function for recursively searching the files

Signed-off-by: Parth Shandilya <parth1989shandilya@gmail.com>

ParthS007 committed:
Add Docs viewer with support of monaco editor

Signed-off-by: Parth Shandilya <parth1989shandilya@gmail.com>

ParthS007 commented:
@DivergentEuropeans Can you please review it? Thanks :)

ParthS007 committed:
Added document reading and setting in the state
Added Docview component(WIP)

Signed-off-by: Parth Shandilya <parth1989shandilya@gmail.com>

ParthS007 committed:
Configured webpack for monaco editor

Signed-off-by: Parth Shandilya <parth1989shandilya@gmail.com>

ParthS007 committed:
Added monaco editor in docview component
Made changes in webpack config

Signed-off-by: Parth Shandilya <parth1989shandilya@gmail.com>

1000TurquoisePogs reviewed (changes requested):
These are the things I spotted but I still need to try it out for myself to see the visuals & behavior.
Biggest concerns are to switch sync to async and to switch from js-based css to "css modules"

1000TurquoisePogs commented:
I havent been able to build this yet,
```
> npm run build

> org.zowe.zlux.desktop.documentation.viewer.webclient@1.0.0 build /sgrady/zlux/zlux-app-manager/system-apps/desktop-documentation-viewer/webClient
> webpack --progress --colors

/sgrady/zlux/zlux-app-manager/system-apps/desktop-documentation-viewer/webClient/node_modules/webpack-cli/bin/cli.js:453
                                throw err;
                                ^

Error: Cannot find module 'vs/editor/standalone/browser/quickOpen/gotoLine'
```

ParthS007 committed:
Added Async await for getting the llist of files
Added doc strings
Added support for css modules and migrated from js-based modules

Signed-off-by: Parth Shandilya <parth1989shandilya@gmail.com>

1000TurquoisePogs commented:
This is looking much better. I'll give it a try and inspect further and hopefully approve. Just one note though: This requires node 8+, will not work on node 6, due to use of `async` and `await`. I don't consider this to be a problem since 6 support has ended.

ParthS007 commented:
Yes @1000TurquoisePogs , node 8+ is required. I am currently working on making the resizing of monaco-editor dynamic with the window size. 

Dynamic Change of `language` is working fine with different types of docs as mentioned above.

ParthS007 committed:
Added multi language support for different doc types
Made changes in webpack config

Signed-off-by: Parth Shandilya <parth1989shandilya@gmail.com>
